src/rulesets/Base/fastmath_able.jl

```diff
@@ -37,9 +37,16 @@ let
 
         # Unary complex functions
         @scalar_rule abs(x::Real) sign(x)
-        @scalar_rule abs2(x) 2x
+        @scalar_rule abs2(x::Real) 2x
         @scalar_rule angle(x::Real) Zero()
         @scalar_rule conj(x::Real) One()
+        function frule((Δx,), abs, x::ComplexF64)
```

Comment:
Can this be generalized for `Complex`? I see what I shared was too specific.
```suggestion
        function frule((Δx,), abs, x::Complex)
```